samples/routing/fair-distribution/Core_6/Client/Program.cs

```diff
@@ -83,9 +83,9 @@ static void AddRouting(EndpointConfiguration endpointConfiguration)
         #region Routing
 
         const string server = "Samples.FairDistribution.Server";
-        var routing = endpointConfiguration.Routing();
+        var routing = endpointConfiguration.UseTransport<MsmqTransport>().Routing();
         routing.RouteToEndpoint(typeof(PlaceOrder), server);
-        routing.Mapping.Physical.Add(
+        routing.GetSettings().GetOrCreate<EndpointInstances>().Add(
```

Comment:
I think we should just use the file-based mapping as the most obvious option. As for the extensibility APIs, they are in the place they are most intuitive-- on a feature context because extending the routing should be done via a feature.